ports/renesas-ra/boards/VK-RA6M3: Another New Board.

mbedNoobNinja opened the pull request:
Thanks to the kind help of @TakeoTakahashi2020 & the mentioned intention for a new ports earlier (#10595), this confirms **2-nd** board is also uPy ready.

`ports/renesas-ra/boards/`**`VK-RA6M3`:** Adding another new dev board
`ports/renesas-ra/boards:` Based on new **RA6M3** processor
`ports/renesas-ra:` Part of existing **RA** family.

[Tests.log](https://github.com/micropython/micropython/files/10899301/M3_Tests.log)

TakeoTakahashi2020 commented:
Thanks for your  contribution, @mbedNoobNinja .

![image](https://user-images.githubusercontent.com/74999249/218909274-d0886fdb-bccb-4ff9-a05c-68dd48bc3aab.png)

Could you check this code?
I think `0x100000` should be `0x10000` because 32KB block starts from 0x10000.

TakeoTakahashi2020 commented:
Hello @mbedNoobNinja -San,

> #define MICROPY_HW_BOARD_NAME "VK_RA6M3"

You would be able to use "VK-RA6M3" of the product name for this macro definition as well.

mbedNoobNinja commented:
Hi @TakeoTakahashi2020-San,

I set the name, you suggested, but for some reason I couldn't fix the conflict in `mpconfigport.h`.

mbedNoobNinja committed:
ports/renesas-ra/boards/VK-RA6M3: New Board.

ports/renesas-ra/boards: New processor RA6M3.
ports/renesas-ra: Integration to the existing RA family.

Signed-off-by: mbedNoobNinja <novoltage@gmail.com>

mbedNoobNinja commented:
Hi @dpgeorge I rebased the PR do you consider it is ready for merge or it is need more work?

mbedNoobNinja commented:
**M3** gained **LCD** support. Now the board can take advantage of **[micropython-nano-gui](https://github.com/peterhinch/micropython-nano-gui)** lib. There is also touch support inspired from **[here](https://docs.circuitpython.org/projects/focaltouch/en/latest/_modules/adafruit_focaltouch.html)** but the user have to process the dots of its own (i.e. manually call the methods **touched()** / **touches()** of the LCD calss in the machine modue & see if the returned points are from the area of interest)

mbedNoobNinja commented:
**M3** gained **CAM** support. Now the board can take advantage of **[OV7725](https://www.arducam.com/wp-content/uploads/2015/08/OV7725-1.jpg)**. Even more the CAM can now be displayed (or not) on the **LCD** with the CAM calss methods **CAM.show_start(LCD)** and **CAM.show_stop()**. Snapshots are also possible to be saved in the internal FLASH which subsequently can be downloaded on PC with **mpremote**. Methods are **CAM.jpeg('filename')** and **CAM.yuyv('filename')**. There are also a various effects that can be applied black and white, sepia etc.

projectgus commented:
This is an automated heads-up that we've just merged a Pull Request
that removes the STATIC macro from MicroPython's C API.

See https://github.com/micropython/micropython/pull/13763

A search suggests this PR might apply the STATIC macro to some C code. If it
does, then next time you rebase the PR (or merge from master) then you should
please replace all the `STATIC` keywords with `static`.

Although this is an automated message, feel free to @-reply to me directly if
you have any questions about this.
